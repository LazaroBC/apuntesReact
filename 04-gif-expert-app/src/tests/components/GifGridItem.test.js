import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { GifGridItem } from '../../components/GifGridItem';

/*
    1.- Enzyme
    2.- Enzyme to json
    3.- Debe de mostrar el componente correctamente

    * Shallow
    * Wrapper
    * Wrapper .toMatchSnapshot()

*/

describe('Pruebas es <GifGridItem />', () => {

    const title = 'Un t√≠tulo';
    const url = 'https://localhost/algo.jpg';

    let wrapper = shallow(
        <GifGridItem
            title={title}
            url={url}
        />);

    test('Debe de mostrar <GifGridItem /> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de tener un parrafo con el title', () => {

        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);

    })

    test('Debe de tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');
        // console.log(img.props().src);
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    })

    test('Debe de tener animate__fadeIn', () => {

        const div = wrapper.find('div');
        // console.log(div.prop('className').includes('animate__fadeIn'));
        expect(div.prop('className').includes('animate__fadeIn')).toBe(true);
        
    })
    

})