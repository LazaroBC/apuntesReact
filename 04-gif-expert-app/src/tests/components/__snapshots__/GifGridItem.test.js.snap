// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas es <GifGridItem /> Debe de mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
